assets/src/dashboard/app/views/editorSettings/index.js

```diff
@@ -21,53 +21,142 @@ import { __ } from '@wordpress/i18n';
 /**
  * External dependencies
  */
-import { useCallback, useContext, useEffect } from 'react';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
 
 /**
  * Internal dependencies
  */
+import { Layout } from '../../../components';
 import { ApiContext } from '../../api/apiProvider';
+import { useConfig } from '../../config';
+import { PageHeading } from '../shared';
 import GoogleAnalyticsSettings from './googleAnalytics';
-import { Wrapper, Header, Heading, Main } from './components';
+import { Wrapper, Main } from './components';
+import PublisherLogoSettings from './publisherLogo';
 
 function EditorSettings() {
   const {
     actions: {
       settingsApi: { fetchSettings, updateSettings },
+      mediaApi: { uploadMedia, fetchMediaById },
     },
     state: {
-      settings: { googleAnalyticsId },
+      settings: { activePublisherLogoId, googleAnalyticsId, publisherLogoIds },
+      media: { isLoading: isMediaLoading, uploadedMediaIds, publisherLogos },
     },
   } = useContext(ApiContext);
 
+  const { capabilities: { canUploadFiles } = {}, maxUpload } = useConfig();
+
+  const [mediaError, setMediaError] = useState('');
+  /**
+   * WP settings references publisher logos by ID.
+   * We must retrieve the media for those ids from /media when present
+   * Summarized below:
+   * FETCH - fetch settings to know what IDs to load media for
+   * then fetch media when publisherLogoIds are in state with useEffect
+   * ADD - first upload new logo to media. use the new ID(s) to update settings.
+   * REMOVE - send id to remove to settings. Not deleting media, just removing from logos.
+   */
+
   useEffect(() => {
     fetchSettings();
   }, [fetchSettings]);
 
-  const handleCompleteUpdateId = useCallback(
-    ({ newGoogleAnalyticsId }) => {
-      const updatedSettings = {
-        googleAnalyticsId:
-          typeof newGoogleAnalyticsId === 'string' && newGoogleAnalyticsId,
-      };
+  useEffect(() => {
+    if (uploadedMediaIds.length > 0) {
+      updateSettings({ publisherLogoIds: uploadedMediaIds });
+    }
+  }, [updateSettings, uploadedMediaIds]);
```

Comment:
`uploadedMediaIds` is only set when new files are added as publisher logos. It's cleared when the upload starts and the Ids comeback on success. This shows us the proper id values to store as publisher logos in the settings API